kivy/core/video/video_ffpyplayer.py

```diff
@@ -262,8 +267,14 @@ def _next_frame_run(self, ffplayer):
         # wait until loaded or failed, shouldn't take long, but just to make
         # sure metadata is available.
         while not self._ffplayer_need_quit:
-            if ffplayer.get_metadata()['src_vid_size'] != (0, 0):
-                break
+            if self._is_stream:
+                # check if src_pix_fmt instead of src_vid_size, which is (0, 0)
+                # at least with rtsp streams
+                if ffplayer.get_metadata()['src_pix_fmt'] != '':
```

Comment:
We can probably just always check that the pixel format is known rather than conditionally checking either the pix fmt or the size. We don't actually use either, this is just a way to check that things are ready.

In the past we may have needed to know the size to finish setting up the player widget. But, we seemingly only update it after the fixed actual image has arrived, so there may not even be a reason to wait until the metadata is populated. As far as I can tell from a quick skim, we don't use the metadata anywhere, except the duration and that only effects the seeking so if it has not been populated it's just zero. 

So we could potentially drop this waiting loop altogether. But it needs to be tested.

Reply:
I removed the wait loop entirely. There's already a loop above checking src_pix_fmt which is used to determine whether to use the custom GLSL shader.